File: src/main/java/itmo/is/project/dto/module/dock/DockModuleDto.java
package itmo.is.project.dto.module.dock;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;

import java.util.List;

public record DockModuleDto(

        @Schema(example = "1")
        @JsonProperty(value = "id", required = true)
        Integer id,

        @JsonProperty(value = "blueprint", required = true)
        DockModuleBlueprintDto blueprint,

        @JsonProperty(value = "dockingSpots", required = true)
        List<DockingSpotDto> dockingSpots
) {
}

File: src/main/kotlin/itmo/isproject/dock/dto/DockModuleDto.kt
package itmo.isproject.dock.dto

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema

data class DockModuleDto(

    @Schema(example = "1")
    @JsonProperty(value = "id", required = true)
    val id: Int?,

    @JsonProperty(value = "blueprint", required = true)
    val blueprint: DockModuleBlueprintDto,

    @JsonProperty(value = "dockingSpots", required = true)
    val dockingSpots: List<DockingSpotDto>
)
